hazelcast-jet-core/src/main/java/com/hazelcast/jet/impl/JobCoordinationService.java

```diff
@@ -998,4 +1012,53 @@ NodeEngineImpl nodeEngine() {
     void assertOnCoordinatorThread() {
         assert IS_JOB_COORDINATOR_THREAD.get() : "not on coordinator thread";
     }
+
+    @Override
+    public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollectionContext context) {
+        descriptor = descriptor.withTag(MetricTags.MODULE, "jet");
+        stats.provideDynamicMetrics(descriptor.copy(), context);
+    }
+
+    private static class Stats implements DynamicMetricsProvider {
+
+        private final AtomicInteger jobSubmitted = new AtomicInteger();
+        private final AtomicInteger jobCompletedSuccessfully = new AtomicInteger();
+        private final AtomicInteger jobCompletedWithFailure = new AtomicInteger();
+        private final AtomicInteger executionStarted = new AtomicInteger();
+        private final AtomicInteger executionTerminated = new AtomicInteger();
+
+        void jobSubmitted() {
+            jobSubmitted.incrementAndGet();
+        }
+
+        void jobCompleted(Throwable error) {
+            if (error == null) {
+                jobCompletedSuccessfully.incrementAndGet();
+            } else {
+                jobCompletedWithFailure.incrementAndGet();
+            }
+        }
+
+        public void executionStarted() {
+            executionStarted.incrementAndGet();
+        }
+
+        void executionTerminated() {
+            executionTerminated.incrementAndGet();
+        }
+
+        @Override
+        public void provideDynamicMetrics(MetricDescriptor descriptor, MetricsCollectionContext context) {
+            context.collect(descriptor, "jobs.submitted", ProbeLevel.INFO, ProbeUnit.COUNT,
```

Comment:
these are not dynamic metrics. You can just annotate them with `@Probe` and then collect the object directly

Reply:
Ok, done in next push.